src/cipher.rs

```diff
@@ -108,6 +110,45 @@ pub extern "C" fn rustls_certified_key_build(
     }
 }
 
+/// Create a copy of the rustls_certified_key with the given OCSP response data
+/// as DER encoded bytes. The OCSP response may be given as NULL to clear any
+/// possibly present OCSP data from the cloned key.
+/// The cloned key is independant from its original and needs to be freed
+/// by the application.
+#[no_mangle]
+pub extern "C" fn rustls_certified_key_clone_with_ocsp(
+    key: *const rustls_certified_key,
+    ocsp_response: *const rustls_slice_bytes,
+    cloned_key_out: *mut *const rustls_certified_key,
+) -> rustls_result {
+    ffi_panic_boundary! {
+        let cloned_key_out: &mut *const rustls_certified_key = unsafe {
+            match cloned_key_out.as_mut() {
+                Some(c) => c,
+                None => return NullParameter,
+            }
+        };
+        let certified_key: Arc<CertifiedKey> = unsafe {
+            match (key as *const CertifiedKey).as_ref() {
+                Some(c) => arc_with_incref_from_raw(c),
+                None => return NullParameter,
+            }
+        };
+        let mut nkey = certified_key.deref().clone();
+        if !ocsp_response.is_null() {
+            unsafe {
+                nkey.ocsp = Some(Vec::from(slice::from_raw_parts((*ocsp_response).data, (*ocsp_response).len)));
+            }
+        }
+        else {
+            nkey.ocsp = None;
+        }
+        let ncertified_key = Arc::into_raw(Arc::new(nkey)) as *const _;
+        *cloned_key_out = ncertified_key;
```

Comment:
Let's make these one line:

```suggestion
        *cloned_key_out = Arc::into_raw(Arc::new(nkey)) as *const _;
```